server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CacheContainerResource.java

```diff
@@ -386,7 +386,7 @@ public void registerChildren(ManagementResourceRegistration resourceRegistration
         super.registerChildren(resourceRegistration);
 
         // child resources
-        resourceRegistration.registerSubModel(new HealthResource(pathManager));
+        resourceRegistration.registerSubModel(new HealthResource(pathManager, runtimeRegistration));
```

Comment:
I tried to do that, but it resulted in a nasty NPE:

```
[Host Controller] 08:21:55,468 ERROR [org.jboss.as.controller.management-operation] (Controller Boot Thread) WFLYCTL0013: Operation ("add") failed - address: ([
[Host Controller]     ("profile" => "clustered"),
[Host Controller]     ("subsystem" => "datagrid-infinispan"),
[Host Controller]     ("cache-container" => "clustered"),
[Host Controller]     ("health" => "HEALTH")
[Host Controller] ]): java.lang.NullPointerException
[Host Controller] 	at org.jboss.as.controller.OperationContextImpl.createResourceInternal(OperationContextImpl.java:941)
[Host Controller] 	at org.jboss.as.controller.OperationContextImpl.createResource(OperationContextImpl.java:935)
[Host Controller] 	at org.jboss.as.controller.AbstractAddStepHandler.createResource(AbstractAddStepHandler.java:198)
[Host Controller] 	at org.jboss.as.controller.AbstractAddStepHandler.createResource(AbstractAddStepHandler.java:184)
[Host Controller] 	at org.jboss.as.controller.AbstractAddStepHandler.execute(AbstractAddStepHandler.java:144)
[Host Controller] 	at org.jboss.as.controller.AbstractOperationContext.executeStep(AbstractOperationContext.java:890)
[Host Controller] 	at org.jboss.as.controller.AbstractOperationContext.processStages(AbstractOperationContext.java:659)
[Host Controller] 	at org.jboss.as.controller.AbstractOperationContext.executeOperation(AbstractOperationContext.java:370)
[Host Controller] 	at org.jboss.as.controller.OperationContextImpl.executeOperation(OperationContextImpl.java:1329)
[Host Controller] 	at org.jboss.as.controller.ModelControllerImpl.boot(ModelControllerImpl.java:493)
[Host Controller] 	at org.jboss.as.controller.AbstractControllerService.boot(AbstractControllerService.java:387)
[Host Controller] 	at org.jboss.as.controller.AbstractControllerService.boot(AbstractControllerService.java:349)
[Host Controller] 	at org.jboss.as.host.controller.DomainModelControllerService.boot(DomainModelControllerService.java:694)
[Host Controller] 	at org.jboss.as.controller.AbstractControllerService$1.run(AbstractControllerService.java:299)
[Host Controller] 	at java.lang.Thread.run(Thread.java:745)
[Host Controller] 
[Host Controller] 08:21:55,472 FATAL [org.jboss.as.host.controller] (Controller Boot Thread) WFLYHC0034: Host Controller boot has failed in an unrecoverable manner; exiting. See previous messages for details.
[Host Controller] 08:21:55,473 FATAL [org.jboss.as.host.controller] (Controller Boot Thread) WFLYHC0178: Aborting with exit code 99
```

It looks like we will have to stay with current version :(